src/models/Collectible.js

```diff
@@ -18,27 +18,35 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+// Constants
+import { ASSET_TYPES } from 'constants/assetsConstants';
+
 // Types
 import type { Chain, ChainRecord } from 'models/Chain';
 
 
 export type Collectible = {
+  tokenType: typeof ASSET_TYPES.COLLECTIBLE,
+  contractAddress: string,
   id: string,
+  chain: Chain,
   name: string,
   description: ?string,
   icon: ?string,
   iconUrl: ?string,
   image: ?string,
   imageUrl: ?string,
-  contractAddress: string,
-  tokenType: string,
-  symbol?: void,
-  decimals?: void,
-  chain: Chain,
 
   // Legacy NFTs are minted pre ERC-721 (e.g. cryptokitties) and require using `transfer` method instead of one
   // inferred from ABI.
   isLegacy: boolean,
+
+  // Cooperation with AssetData/TokenData model
+  token?: void,
```

Comment:
Do we really need that? Is there any way we can get rid of `token`? Having it has many down sides that might end up in wrong mappings where `symbol` can be missing because of `token` and we only look for `symbol`.

Reply:
This is only to clarify the typing in the components we already use. Cleaning that up is a matter of large tech debt PR.